de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/wizard/CsvFileReqTypeSelectionPage.java

```diff
@@ -0,0 +1,320 @@
+		labelDataEndNumber.setText(DATA_LINE_END_LABEL);
+
+		dataNumberEndField = new Text(propertiesComposite, SWT.SINGLE | SWT.BORDER);
+		data = new GridData(GridData.GRAB_HORIZONTAL);
+		data.widthHint = WITH_TEXT;
+		data.horizontalAlignment = SWT.END;
+		dataNumberEndField.setLayoutData(data);
+		dataNumberEndField.addModifyListener(this);
+
+	}
+
+	/**
+	 * Create a tree viewer with filters to show only relevant tree elements for
+	 * CSV import /export
+	 */
+	protected void createTreeViewer() {
+		TreeViewer treeViewer = createTreeUI();
+		VirSatFilteredWrappedTreeContentProvider filteredCp = (VirSatFilteredWrappedTreeContentProvider) treeViewer
+				.getContentProvider();
+		filteredCp.addClassFilter(CategoryAssignment.class);
+		filteredCp.addClassFilter(ArrayInstance.class);
+		filteredCp.addClassFilter(ComposedPropertyInstance.class);
+		filteredCp.addStructuralElementIdFilter(
+				RequirementsConfigurationCollection.FULL_QUALIFIED_STRUCTURAL_ELEMENT_NAME);
+
+	}
+
+	@Override
+	public boolean isSelectionValid() {
+		Object selection = getSelection();
+		if (selection instanceof CategoryAssignment) {
+			return ((CategoryAssignment) selection).getType().getFullQualifiedName()
+					.equals(RequirementsConfiguration.FULL_QUALIFIED_CATEGORY_NAME)
+					|| ((CategoryAssignment) selection).getType().getFullQualifiedName()
+							.equals(RequirementType.FULL_QUALIFIED_CATEGORY_NAME);
+		}
+		return false;
+	}
+
+	@Override
+	protected String[] getSupportedFileEndings() {
+		return FILE_EXTENSIONS;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see de.dlr.sc.virsat.uiengine.ui.wizard.AImportExportPage#isComplete()
+	 */
+	@Override
+	public boolean isComplete() {
+
+		if (isDestinationSelected && isCurrentPage()) {
+			if (!headerNumberField.getText().equals("") & !seperatorField.getText().equals("") 
+				& !dataNumberStartField.getText().equals("")) {
+
+				final String destination = getDestination();
+	
+				if (isSelectionValid()) {
+					CategoryAssignment selection = (CategoryAssignment) getSelection();
+					Repository repository = VirSatResourceSet.getVirSatResourceSet(selection).getRepository();
+					ActiveConceptHelper activeConceptHelper = new ActiveConceptHelper(repository);
+					Concept activeReqConcept = activeConceptHelper.getConcept(de.dlr.sc.virsat.model.extension.requirements.Activator.getPluginId());
+			
+					try {
+						wizard.getReader().setSeparator(getSeparator());
+						wizard.getReader().setHeaderLine(getHeaderLineNumber());
+						csvHeader = wizard.getReader().readCsvHeadline(destination);
+					} catch (IOException e) {
+						Status status = new Status(Status.ERROR, Activator.getPluginId(),
+								"CatiaImportWizard: Failed to perform import! Selected file not valid!", e);
+						StatusManager.getManager().handle(status, StatusManager.LOG | StatusManager.SHOW);
+					}
+					
+					if (selection.getType().getFullQualifiedName().equals(RequirementType.FULL_QUALIFIED_CATEGORY_NAME)) {

```

Comment:
Actually that's how I had it before... Thats why there was this unsued getter methode left... The problem then is that the page is not properly updated if you back to the first and then back to the second page because the page was already created with the old values. I actually like the push-princeple here... And I don't think a selection change will happen that often?